packages/engine-core/src/library/LibraryEngine.ts

```diff
@@ -38,6 +41,10 @@ function isQueryEvent(event: QueryEngineEvent): event is QueryEngineQueryEvent {
   return event['item_type'] === 'query' && 'query' in event
 }
 function isPanicEvent(event: QueryEngineEvent): event is QueryEnginePanicEvent {
+  if ('span' in event) {
```

Comment:
Wait, so we have a situations where `span` is in the event and the message is `PANIC` at the same time? When does it happen? If it does, are we really sure that the right thing to do here is ignoring panic?